src/for-mobx-react-query/SpreadoSetupImpl.ts

```diff
@@ -0,0 +1,65 @@
+import {QueryClient} from 'react-query';
+import {generateSpreadKey, SpreadoSetupBase, SpreadoSetupOptions} from '../core';
+import {getSpreadIn, setSpreadOut, SpreadoMobXStore, useSpreadIn, useSpreadOut} from '../mobx';
+
+interface Options extends SpreadoSetupOptions {
+  store?: SpreadoMobXStore;
+  enableAutoSpread?: boolean;
+  queryClient: QueryClient;
+}
+
+interface InternalOptions extends Options {
+  store: SpreadoMobXStore;
+}
+
+export class SpreadoSetupForMobXReactQuery extends SpreadoSetupBase {
+  private options: InternalOptions;
+  private useSpreadOutCounter: Record<string, number> = {};
+
+  constructor(options: Options) {
+    super(options);
+    this.options = {
+      store: options.store ?? new SpreadoMobXStore(),
+      ...options,
+    };
+
+    const {enableAutoSpread} = this.options;
+    if (enableAutoSpread) {
+      this.autoSpreadOutQueries();
+    }
+  }
+
+  autoSpreadOutQueries() {
```

Comment:
Let's name this method in a same way as `for-redux-react-query/SpreadoSetupImpl.ts`:

```suggestion
  enableQueriesAutoSpreadOut() {
```